firebase-database/src/main/java/com/google/firebase/database/core/utilities/PushIdGenerator.java

```diff
@@ -57,6 +65,38 @@ public static synchronized String generatePushChildName(long now) {
     return result.toString();
   }
 
+  public static final String nextAfter(String key) {
+    Validation.validateNullableKey(key);
+    Integer num = tryParseInt(key);
+    if (num != null) {
+      return String.valueOf(num + 1);
```

Comment:
This looks like it could trigger an integer overflow. You could probably just use a long here, unless there are some specifications for valid keys that I am not aware of.

Reply:
Using a long should be fine, the limit on keys is 768 bytes:

https://firebase.google.com/docs/database/usage/limits#data_tree

Reply:
The only other place we're using tryParseInt is in:

https://github.com/firebase/firebase-android-sdk/blob/e65f4876ff61b5791986c249b21cb094c37bd29e/firebase-database/src/main/java/com/google/firebase/database/snapshot/ChildrenNode.java#L134

Where it seems to be used to decide the number of results in an array. I think we should be able to change the method to `tryParseLong` without risk.

Reply:
Upon further consideration, I think we should handle the overflow here. In our public docs, we assume at most 32-bit integer keys for ordering semantics:

https://firebase.google.com/docs/database/web/lists-of-data

I wonder if it's reasonable to return `MIN_PUSH_CHAR` here since that's the next string in our documented key ordering.